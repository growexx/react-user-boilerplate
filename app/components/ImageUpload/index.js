/**
 *
 * ImageUpload
 *
 */

import React from 'react';
import { Upload } from 'antd';
import ImgCrop from 'antd-img-crop';
import 'antd/es/modal/style';
import 'antd/es/slider/style';
import { API_ENDPOINTS } from 'containers/constants';

class ImageUpload extends React.Component {
  state = {
    fileList: [],
  };

  onChange = ({ fileList: newFileList }) => {
    this.setState({
      fileList: newFileList,
    });
  };

  onPreview = async file => {
    let src = file.url;
    if (!src) {
      src = await new Promise(resolve => {
        const reader = new FileReader();
        reader.readAsDataURL(file.originFileObj);
        reader.onload = () => resolve(reader.result);
      });
    }
    const image = new Image();
    image.src = src;
    const imgWindow = window.open(src);
    imgWindow.document.write(image.outerHTML);
  };

  render() {
    return (
      <ImgCrop rotate>
        <Upload
          action={API_ENDPOINTS.IMAGE_UPLOAD}
          listType="picture-card"
          fileList={this.state.fileList}
          onChange={this.onChange}
          onPreview={this.onPreview}
          maxCount={1}
        >
          {this.state.fileList.length < 1 && '+ Upload'}
        </Upload>
      </ImgCrop>
    );
  }
}

ImageUpload.propTypes = {};

export default ImageUpload;
