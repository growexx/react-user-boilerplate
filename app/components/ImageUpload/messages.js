/*
 * ImageUpload Messages
 *
 * This contains all the text for the ImageUpload component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ImageUpload';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ImageUpload component!',
  },
});
