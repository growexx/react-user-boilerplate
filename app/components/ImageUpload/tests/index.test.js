/**
 *
 * Tests for ImageUpload
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent, act, wait } from 'react-testing-library';
import 'jest-dom/extend-expect';
import { IntlProvider } from 'react-intl';
import ImageUpload from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
jest.mock('antd-img-crop', () => {
  const ComponentToMock = Children => <div>{Children.children}</div>;
  return ComponentToMock;
});
describe('<ImageUpload />', () => {
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ImageUpload />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
  it('Should onchange the file upload', async () => {
    global.URL.createObjectURL = jest.fn();
    global.Image = jest.fn().mockImplementation(() => ({}));
    global.open = jest.fn().mockImplementation(() => ({
      document: {
        write: jest.fn(),
      },
    }));
    HTMLCanvasElement.prototype.getContext = () => jest.fn();
    const { container, getByTitle } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ImageUpload />
      </IntlProvider>,
    );
    const file = new File(['(⌐□_□)'], 'test.png', {
      type: 'image/png',
      lastModified: Date.now(),
    });
    const hiddenFileInput = container.querySelector('input');
    await act(async () => {
      fireEvent.change(hiddenFileInput, { target: { files: [file] } });
    });
    await wait(async () => {
      expect(container.querySelector('a')).toBeInTheDocument();
    });
    fireEvent.click(getByTitle('Preview file'));
    expect(hiddenFileInput.files[0]).toStrictEqual(file);
  });
});
