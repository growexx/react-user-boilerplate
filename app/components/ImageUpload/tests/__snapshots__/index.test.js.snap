// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUpload /> Should render and match the snapshot 1`] = `
<div>
  <span
    class="ant-upload-picture-card-wrapper"
  >
    <div
      class="ant-upload-list ant-upload-list-picture-card"
    >
      <div
        class="ant-upload ant-upload-select ant-upload-select-picture-card"
      >
        <span
          class="ant-upload"
          role="button"
          tabindex="0"
        >
          <input
            accept=""
            style="display: none;"
            type="file"
          />
          + Upload
        </span>
      </div>
    </div>
  </span>
</div>
`;
